pkg/action/action.go

```diff
@@ -97,6 +97,19 @@ type Configuration struct {
 	Log func(string, ...interface{})
 }
 
+// chartNeedsAPIVersions returns true if any of the Chart's templates reference "APIVersions".
+// If it returns false, it is safe to omit Capabilities.APIVersions from rendering values.
+func chartNeedsAPIVersions(chart *chart.Chart) bool {
```

Comment:
The implementation here: string search of the templates is a little brittle IMHO (at least the `tpl` function exists, which can dynamically execute templates

Let me try and see if there is an alternative.

Reply:
Thanks @gjenkins8. I would definitely prefer something more precise, but don't yet know a better way.
One of my colleagues had an idea: could the implementation of `APIVersions` change to be a function that when referenced, fetches the list or returns a cached copy from previous calls?

Reply:
https://kubernetes.io/docs/concepts/overview/kubernetes-api/#aggregated-discovery perhaps (I didn't check the details, but from the tin seems to be a more efficient way of fetching API versions)

Reply:
Following up on the suggested alternatives...

> Could the implementation of APIVersions change to be a function that when referenced, fetches the list or returns a cached copy from previous calls?

After taking a closer look, even if viable, this would require a breaking change to the `APIVersions` field within the public `Capabilities` struct.

> https://kubernetes.io/docs/concepts/overview/kubernetes-api/#aggregated-discovery

It looks like [ServerGroupsAndResources](https://github.com/kubernetes/client-go/blob/v0.31.0/discovery/discovery_client.go#L471) (used in [GetVersionSet](https://github.com/helm/helm/blob/v3.16.1/pkg/action/action.go#L319)) uses aggregated discovery by default if the provided client supports it. I'll try to evaluate the amount of effort it would take to use a compatible client.

Reply:
I'm almost positive the Helm library already supports using aggregated discovery clients. It just depends on the implementation of RESTClientGetter.ToDiscoveryClient. If your ToDiscoveryClient returns a client using [k8s.io/client-go/discovery](https://pkg.go.dev/k8s.io/client-go/discovery) v0.30 and up, it should use aggregated discovery OOTB.

What I'm not sure about is how the Helm CLI implements RESTClientGetter. I haven't been able to find that anywhere in the code so far.

Reply:
I confirmed that the discovery client used by the latest version of the Helm CLI supports aggregated discovery.
So I think this PR is not necessary! Thanks @gjenkins8 again for your feedback and suggestions.